### PR TITLE
[ISSUE #5034] add ut for package com.alibaba.nacos.client.config.http

### PR DESCRIPTION
**Pull request by shalk**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

fix [ISSUE #5034] add ut for package com.alibaba.nacos.client.config.http

line coverage 59%   

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Review by JackSun-Developer — APPROVED** [review]
Looks good.

File: client/src/test/java/com/alibaba/nacos/client/config/http/ServerHttpAgentTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    @Test
+    public void testHttpGet() throws Exception {
+        Properties properties = new Properties();
+        properties.put(PropertyKeyConst.SERVER_ADDR, "https://httpbin.org");
+        properties.put(PropertyKeyConst.CONTEXT_PATH, "/");
+        final ServerHttpAgent serverHttpAgent = new ServerHttpAgent(properties);
+        final HttpRestResult<String> result = serverHttpAgent
+                .httpGet("/get", new HashMap<String, String>(), new HashMap<String, String>(), "UTF-8", 5000);
+        
+        Assert.assertEquals(200, result.getCode());
```
Comment (JackSun-Developer) [inline client/src/test/java/com/alibaba/nacos/client/config/http/ServerHttpAgentTest.java:77]: 这个返回200在进行单测的时候并不能保证吧，因为服务端可能都没有启动起来。

**Review by JackSun-Developer — CHANGES_REQUESTED** [review]
单测不应该走网络，应当是纯粹内存操作。个人认为正确的做法是，Mock出来一个NacosRestTemplate NACOS_RESTTEMPLATE（完全搞成内存操作），通过反射塞进ServerHTTPAgent类里面。然后，运行httpGet，assert一下NACOS_RESTTEMPLATE.get被调用过。

File: client/src/test/java/com/alibaba/nacos/client/config/http/ServerHttpAgentTest.java (hunk 1 above)
Comment (shalk) [reply client/src/test/java/com/alibaba/nacos/client/config/http/ServerHttpAgentTest.java:77]: 当前地址是 https://httpbin.org/  托管在github page上，长期运行的

**Comment by shalk** [comment]
> 单测不应该走网络，应当是纯粹内存操作。个人认为正确的做法是，Mock出来一个NacosRestTemplate NACOS_RESTTEMPLATE（完全搞成内存操作），通过反射塞进ServerHTTPAgent类里面。然后，运行httpGet，assert一下NACOS_RESTTEMPLATE.get被调用过。

按讨论，我们就先这样吧

File: client/src/test/java/com/alibaba/nacos/client/config/http/ServerHttpAgentTest.java (hunk 1 above)
Comment (KomachiSion) [reply client/src/test/java/com/alibaba/nacos/client/config/http/ServerHttpAgentTest.java:77]: Skip this case first if this `static final` field is hard to mock.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**